### PR TITLE
Support multiple output files

### PR DESCRIPTION
**Pull request by ErikSchierboom**
This PR adds support for multiple output files.

See exercism/development-environment#34

### CODE REVIEW

File: lib/tooling_invoker/invoke_locally.rb [hunk 1]
```diff
@@ -31,7 +31,12 @@ def call
         cmd: cmd,
         exit_status: exit_status
       }
-      job.result = File.read("#{output_dir}/#{job.results_filepath}")
+
+      job.output_files = {}
+      job.output_filepaths.each do |output_filepath|
+        job.output_files[output_filepath] = File.read("#{output_dir.job_dir}/#{output_filepath}")
+      end
```
Comment (iHiD) [inline lib/tooling_invoker/invoke_locally.rb:38]: Slight suggested refactor:

```suggestion
      job.output_files = job.output_filepaths.each.with_object({}) do |output_filepath, hash|
        hash[output_filepath] = File.read("#{output_dir.job_dir}/#{output_filepath}")
      end
```

File: lib/tooling_invoker/representer_job.rb [hunk 2]
```diff
@@ -20,8 +20,13 @@ def tool
...
 
-    def parsed_result
-      result
+    def output
+      {
+        "representation.txt": output_files["representation.txt"],
+        "mapping.json": JSON.parse(output_files["mapping.json"])
+      }
+    rescue StandardError
+      {}
```
Comment (iHiD) [inline lib/tooling_invoker/representer_job.rb:29]: I feel like we've got a bit of indirection going on here now, which took me a bit of time to get my head around. Having the method setting output_files on job, then them being converted in each job via this.

How about instead we have a writer method on each of these jobs (called `output=`) which takes the hash of filenames and raw files, and then mutates them before storing them? So we don't store them as `output_files`, just as `@output`, and then we can have `attr_reader :output` on the job.rb?

Comment (ErikSchierboom) [reply lib/tooling_invoker/representer_job.rb:29]: I like that. I re-used the indirection that was already there, but the above seems like a better approach.

File: lib/tooling_invoker/invoke_docker.rb [hunk 3]
```diff
@@ -35,8 +35,8 @@ def call
         exit_status: json['exit_status']
       }
 
-      job.result = json['result']
-      job.status = job.result ? 200 : 400
+      job.output = json['output_files']
```
Comment (iHiD) [inline lib/tooling_invoker/invoke_docker.rb:38]: @ErikSchierboom I would like this to also use the `output_filepaths` else we could end up with a situation where a file gets passed through via this route, but not via the runc route. I'll merge but could you add that here, and add a comment explaining, in a separate PR pls?

Comment (ErikSchierboom) [reply lib/tooling_invoker/invoke_docker.rb:38]: Will do

**Pull request merged**